frontend/packages/dev-console/src/components/import/import-types.ts

```diff
@@ -96,6 +97,7 @@ export interface GitImportFormData {
   serverless?: ServerlessData;
   pipeline?: PipelineData;
   image: ImageData;
+  runtimeIcon?: string;
```

Comment:
Yeah I do not like this as well! The problem is that in `create-knative-utils.ts` the types GitImportFormData and DeployImageFormData was used together and I want to extract runtimeIcon from them. And that works only if the property is part of both types.

https://github.com/openshift/console/blob/c116d89a8aa71c87f6aee80b5c66ca15d5f392c8/frontend/packages/knative-plugin/src/utils/create-knative-utils.ts#L11-L35

I could also change `create-knative-utils.ts` to to something like this ... but this looks much more confusing. What do you think? :thinking: 

```diff
 export const getKnativeServiceDepResource = (
-  formData: GitImportFormData | DeployImageFormData,
+  formData: (GitImportFormData & { runtimeIcon: never }) | DeployImageFormData,
   imageStreamUrl: string,
   imageStreamName?: string,
```

Combing both with an & instead of an | does not work because many callers apply only one of this types.

Reply:
Perhaps we just leave it for now. There are type-guards we could probably use here, but I am unsure it's overly needed. Splitting the type values (ie where you use `: never`) is not going to be a fun maintenance issue. I don't think it overly matters if we have an optional property not used.